finagle-core/src/main/scala/com/twitter/finagle/Exceptions.scala

```diff
@@ -134,10 +134,18 @@ class GlobalRequestTimeoutException(timeout: Duration)
  *
  * [1] https://twitter.github.io/finagle/guide/Names.html
  */
-class NoBrokersAvailableException(val name: String, val baseDtab: Dtab, val localDtab: Dtab)
+class NoBrokersAvailableException(val name: String, val baseDtabFn: () => Dtab, val localDtabFn: () => Dtab)
```

Comment:
It's usually pretty unusual and baseDtab changes this late in the game.  how about we keep `baseDtab` a field until someone needs it?